### PR TITLE
chore: fixes link to the namespace implementation

### LINKED ISSUE #3639: Fixing broken link to the namespace implementation
## Problem

Due to some refactoring going on in go-square, the link to the namespace implementation got broken in [here](https://github.com/celestiaorg/celestia-app/blob/077785c4e8149f2646f7b6969d46e750cdbcd711/specs/src/specs/namespace.md#implementation) which results in markdown linter to error out in [here](https://github.com/celestiaorg/celestia-app/actions/runs/9703899691/job/26783006867?pr=3612#step:4:1007).

## Acceptance Criteria
Fix the broken links.

### PR DESCRIPTION
**Pull request by staheri14**
Closes #3639

### CODE REVIEW

File: specs/src/specs/namespace.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ Among the potential consequences is the _Woods Attack_, as elaborated in this fo
 
 ## Implementation
 
-See [go-square/namespace](https://github.com/celestiaorg/go-square/tree/main/namespace).
+See [go-square/namespace](https://github.com/celestiaorg/go-square/blob/main/shares/namespace.go).
```
Comment (staheri14) [inline specs/src/specs/namespace.md:97]: @rootulp Can you please confirm whether this is the correct link?

Comment (rootulp) [reply specs/src/specs/namespace.md:97]: I fixed it slightly differently here: https://github.com/celestiaorg/celestia-app/pull/3635/commits/62c3cd6eb16e02a023da6b688f2d3320925feee3 by using a permalink to go-square/namespace (basically the copy of code before the refactor). What you've done here also works.

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]
thanks!

**Comment by rootulp** [comment]
lol I should've pushed harder for permalinks b/c now the link is broken again b/c we renamed `shares` => `share` on main

**Pull request merged**